new file mode 100644
index 000000000000..ddb8459a760c
--- /dev/null
+++ b/arch/arm/boot/dts/rk3188-ugoos-ut2.dts
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2020 Alex Bee <knaerzche@gmail.com>
+ */
+
+/dts-v1/;
+#include "rk3188-rbox-cs968.dts"
+
+/ {
+	model = "Ugoos UT2";
+	compatible = "radxa,rock", "rockchip,rk3188";
+
+	aliases {
+		/delete-property/ mshc2;
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		sleep {
+			label = "red";
+			gpios = <&gpio0 RK_PA2 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+	};
+
+	/delete-node/ sdio-pwrseq;
+
+
+
+};
+
+/delete-node/ &mmc1;
+
+&emac {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&emac_xfer>, <&emac_mdio>, <&phy_int>;
+
+	phy = <&phy0>;
+	phy-supply = <&vcc_rmii>;
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <RK_PD2 IRQ_TYPE_LEVEL_LOW>;
+	};
+};
+
+//&mmc0 {
+//	status = "okay";
+//	pinctrl-names = "default";
+//	pinctrl-0 = <&sd0_clk>, <&sd0_cmd>,
+//		    <&sd0_cd>, <&sd0_bus4>;
+//	vmmc-supply = <&vcc_sd0>;
+//	bus-width = <4>;
+//	cap-mmc-highspeed;
+//	cap-sd-highspeed;
+//};
+
+&pinctrl {
+	lan8720a  {
+		phy_int: phy-int {
+			rockchip,pins = <3 RK_PD2 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
+};

